### PR TITLE
Update Astro product IA

### PR DESCRIPTION
**Pull request by jwitz**
Resolves https://github.com/astronomer/docs/issues/1227
Resolves https://github.com/astronomer/docs/issues/1224

Two comments for this PR:

- I'm playing with the idea of having Runtime docs in multiple menus (Release notes is in both the Runtime and the Release Notes menus now)
- I couldn't find an appropriate home for the limitation about the Celery executor.. we might need our "Migrate DAGs" doc for that one.

### CODE REVIEW

**Review by kwcanuck — APPROVED** [review]

**Review by paolaperaza — CHANGES_REQUESTED** [review]
Hey Jake -- let's chat about this quick before you merge. From the top, I'm seeing a redundancy in `Astro Runtime` being the name of release notes and the name of the top-level menu:

image

We need to either:
- Get rid of `Release Notes` as a top-level navigation
- Not include Astro Runtime release notes in the Astro Runtime menu

I'm actually not super convinced we _need_ an Astro Runtime menu? Anything wrong with keeping these docs in `Reference` (which I agree, we should clean up)

File: sidebarsAstro.js [hunk 1]
```diff
@@ -77,6 +77,16 @@ module.exports = {
         'airflow-alerts',
       ],
     },
+    {
+      type: 'category',
+      label: 'Astro Runtime',
+      items: [
+        'upgrade-runtime',
+        'runtime-image-architecture',
+        'runtime-release-notes',
```
Comment (jwitz) [inline sidebarsAstro.js:86]:
```suggestion
```

File: sidebarsAstro.js [hunk 2]
```diff
@@ -77,6 +77,15 @@ module.exports = {
...
     },
+    {
+      type: 'category',
+      label: 'Astro Runtime',
+      items: [
+        'upgrade-runtime',
+        'runtime-image-architecture',
+        'runtime-version-lifecycle-policy',
+      ],
+    },
```
Comment (jwitz) [inline sidebarsAstro.js:88]:
```suggestion
```

File: sidebarsAstro.js [hunk 3]
```diff
@@ -129,9 +138,6 @@ module.exports = {
       type: 'category',
       label: 'Reference',
       items: [
-        'known-limitations',
-        'runtime-image-architecture',
-        'runtime-version-lifecycle-policy',
         'astro-support',
```
Comment (jwitz) [inline sidebarsAstro.js:141]:
```suggestion
        'astro-support',
        {
          type: 'category',
          label: 'Astro Runtime',
          items: [
          'runtime-image-architecture',
          'runtime-version-lifecycle-policy',
          ],
      },
```

**Comment by jwitz** [comment]
@paolaperaza I was mainly trying to replicate what we had on Software, where we have both an AC and a Runtime menu. Maybe a compromise would be to have a Runtime submenu in the Reference menu? I've implemented that idea here.

**Review by paolaperaza — APPROVED** [review]

**Comment by paolaperaza** [comment]
@jwitz Should we add a redirect from known-limitations to Astro home?

**Pull request merged**